ddtrace/tracer/option.go

```diff
@@ -188,9 +187,15 @@ const maxPropagatedTagsLength = 512
 func newConfig(opts ...StartOption) *config {
 	c := new(config)
 	c.sampler = NewAllSampler()
-	c.agentAddr = resolveAgentAddr()
+	c.agentURL = "http://" + resolveAgentAddr()
 	c.httpClient = defaultHTTPClient()
-
+	if url := internal.AgentURLFromEnv(); url != nil {
+		if url.Scheme == "unix" {
+			c.httpClient = udsClient(url.Path)
```

Comment:
Yeah, I don't think the host portion of the URL matters when connecting over UDS, since we're overriding the HTTP client to connect to the socket in its [dialer](https://github.com/DataDog/dd-trace-go/blob/274246638d1507d3f9bdc9503e5650d90612dcdf/ddtrace/tracer/option.go#L343-L348). See this playground example: https://go.dev/play/p/3cvE8KJEfT4